%YAML:1.0

# flags
use_imu: 0
use_lidar: 1
use_navsat: 0
use_loop: 0
use_adapt: 0
use_navigation: 1

# ros parameters
imu_topic: '/D435i_camera/imu/sample'
lidar_topic: '/velodyne_points'
navsat_topic: '/kitti/oxts/gps/fix'
image0_topic: '/D435i_camera/infra1/image_rect_raw'
image1_topic: '/D435i_camera/infra2/image_rect_raw'
color_topic: '/D435i_camera/color/image_raw'
nav_goal_topic: '/move_base_simple/goal'
result_path: '/home/zoet/Projects/lvio_fusion/result/result.csv'

# cameras parameters
undistort: 0

# camera0 intrinsics
camera0.fx: 347.99755859375
camera0.fy: 347.99755859375
camera0.cx: 320.0
camera0.cy: 240.0

# camera1 intrinsics
camera1.fx: 347.99755859375
camera1.fy: 347.99755859375
camera1.cx: 320.0
camera1.cy: 240.0

# lidar parameters
num_scans: 16           # only for 16, 32, 64
horizon_scan: 380
ang_res_y: 2.0
ang_bottom: 15.1
ground_rows: 8
cycle_time: 0.1036
min_range: 0.2
max_range: 50
deskew: 0
spacing: 0
resolution: 0.2

#imu parameters
acc_n: 0.08             # accelerometer measurement noise standard deviation. #0.2   0.04
gyr_n: 0.004            # gyroscope measurement noise standard deviation.     #0.05  0.004
acc_w: 0.00004          # accelerometer bias random work noise standard deviation.  #0.02
gyr_w: 2.0e-6           # gyroscope bias random work noise standard deviation.     #4.0e-5
g_norm: 9.81007         # gravity magnitude

# camera0 to body
body_to_cam0: !!opencv-matrix
   rows: 4
   cols: 4
   dt: d
   data: [   0,  0,  1, -0,
            -1,  0, -0,  0,
            0 ,-1 , 0 ,-0,
            0 , 0 ,-0 , 1]

# camera1 to body
body_to_cam1: !!opencv-matrix
   rows: 4
   cols: 4
   dt: d
   data: [  0,   0,   1,     0,
           -1,   0,   0, -0.05,
            0,  -1,   0,    -0,
            0,   0,  -0,     1]

# lidar to body
body_to_lidar: !!opencv-matrix
   rows: 4
   cols: 4
   dt: d
   data: [   1,  0,   0, -0.1,
          0,    1,  -0,   0.0,
            -0,   -0,   1,   0.0,
           -0,    0,  -0,       1]

# number of features
num_features: 200
num_features_init: 30
num_features_tracking_bad: 10
num_features_needed_for_keyframe: 100

# backend
windows_size: 3

# navsat
accuracy: 2

# loop
relocator_mode: 0    # none = 0, visual = 1, lidar = 2, visual&&lidar = 3
threshold: 10

# train
ground_truth_path: /home/zoet/Projects/playground/kitti_00_tum_gd.txt
obs_rows: 4
obs_cols: 12

#navigation
grid_width: 1000
grid_height: 1000
grid_resolution: 0.05